coders/qoi.c

```diff
@@ -325,8 +325,11 @@ static Image *ReadQOIImage(const ImageInfo *image_info,
       "UnexpectedEndOfFile",image->filename);
   if (status != MagickFalse) {
     status=SyncAuthenticPixels(image,exception);
-    if (image->ping == MagickFalse)
-      SyncImage(image,exception);
+    if (image->ping != MagickFalse) {
```

Comment:
This `if` block can be moved to after the line that has the following statement `ThrowReaderException(CorruptImageError,"NegativeOrZeroImageSize");`